### DISCUSSION TITLE
What happens with local workers on a symmetric multimaster?

### DISCUSSION THREAD

**Discussion by alex-nl** [buildbot/buildbot#6309] [Q&A]
I'm looking into deploying a symmetric multimaster, as I want to be able to update the master.cfg without having to take our single master down and interrupting running jobs.  If I have two masters running simultaneously with the exact same configuration, they will each define a local worker with the same name. Is this valid? What will happen if multiple builds are scheduled for the local worker? 

EDIT: I've done some experimenting now. This does work, and it looks like it's functionally identical to defining one worker that can run n builds on it, where n is the number of masters that are running. Is that a fair way of thinking about it?

**Comment by alex-nl** [comment]
Somewhat separately, I see in here https://github.com/buildbot/buildbot/discussions/5769 that @tardyp recommends having a separate master for schedulers and reporters. Are there known issues with a purely symmetric masters?

**Comment by jpflori** [comment]
Can I ask how you deal with the port numbers to connect to the masters if you have the exact same config?

**Reply by alex-nl** [reply]
At the time I was experimenting with this, I was running the master processes on different machines (connected to the same database instance).
